packages/service-library/tests/fastapi/test_tracing.py

```diff
@@ -0,0 +1,115 @@
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+
+import os
+from asyncio import AbstractEventLoop
+
+import pytest
+from fastapi import FastAPI
+from pydantic import ValidationError
+from servicelib.fastapi.tracing import setup_opentelemetry_instrumentation
+from settings_library.tracing import TracingSettings
+
+mock_app = FastAPI(title="Opentelemetry example")
+
+
+@pytest.fixture
+def tracing_settings_in(request):
+    return request.param
+
+
+@pytest.fixture()
+def set_and_clean_settings_env_vars(tracing_settings_in):
+    if tracing_settings_in[0]:
+        os.environ["TRACING_OTEL_COLLECTOR_ENDPOINT"] = tracing_settings_in[0]
+    if tracing_settings_in[1]:
+        os.environ["TRACING_OTEL_COLLECTOR_PORT"] = str(tracing_settings_in[1])
+    yield
+    os.environ.pop("TRACING_OTEL_COLLECTOR_ENDPOINT", None)
+    os.environ.pop("TRACING_OTEL_COLLECTOR_PORT", None)
+
```

Comment:
use the monkeypatch for this.


```suggestion
@pytest.fixture()
def set_and_clean_settings_env_vars(monkeypatch: pytest.MonkeyPatch):
    if tracing_settings_in[0]:
        monkeypatch.setenv("TRACING_OTEL_COLLECTOR_ENDPOINT", f"{tracing_settings_in[0]}")
    if tracing_settings_in[1]:
        monkeypatch.setenv("TRACING_OTEL_COLLECTOR_PORT", f"{tracing_settings_in[1]}")
```